docs/modules/ROOT/pages/how-tos/upgrade-v1-v2.adoc

```diff
@@ -0,0 +1,91 @@
+= Upgrade `component-cert-manager` from `v1.x` to `v2.x`
+
+Version 2.x upgrades the underlying `cert-manager` helm chart from `v0.x` to `v1.x`.
+There are some breaking changes in `cert-manager`.
+More information can be found https://github.com/jetstack/cert-manager/releases[in changelog on GitHub] and in the https://cert-manager.io/docs/installation/upgrading/[upgrade documentation].
+
+The upgrade is done with the following steps:
+
+. Compile Commodore catalog with `component-cert-manager` v2.x
+. Fix CRD upgrade
+. Verify cert-manager upgrade
+
+== Prerequisites
+
+. `commodore`
+. `kubectl` or `oc`
+. https://github.com/mikefarah/yq[`yq`] version 4 (alternatively, any editor works)
+
+== Upgrade
+
+. Compile and push cluster catalog with `component-cert-manager` v2.x
+
+. Fix CRD upgrade
```

Comment:
Is that something which could be automated somehow?

Reply:
I'm not sure. We could hack something with [ArgoCD hooks](https://argoproj.github.io/argo-cd/user-guide/resource_hooks/), but it comes with its own pitfalls. We'd have to ensure that the hook is idempotent and compatible for any upgrades within v2.x versions. Even the ServiceAccount executing the Job would need to be granted cluster admin permissions in order to modify CRDs.

Reply:
We could try with hooks. The hooks would need to be of type `PreSync` or in an earlier wave for Argo not to fail before executing the hooks.

The fix should be idempotent and not too hard to keep compatible since it does essentially nothing 😅.

Reply:
Yes, but maybe you need another half-day to a day to engineer a working hook with the cluster-admin permission and docker image it requires to perform the hook

Reply:
If you have a component like this which might be deployed to a huge amount of clusters I don't think it's feasible if you have to manually touch every cluster to upgrade the component. I would find it a valuable time investment to try to automate the upgrade.

Reply:
Started automating the process in #18.